leakybucket_test.go

```diff
@@ -11,27 +11,27 @@ import (
 )
 
 // leakyBuckets returns all the possible leakyBuckets combinations.
-func (s *LimitersTestSuite) leakyBuckets(capacity int64, rate time.Duration, clock l.Clock) []*l.LeakyBucket {
-	var buckets []*l.LeakyBucket
-	for _, locker := range s.lockers(true) {
-		for _, backend := range s.leakyBucketBackends() {
-			buckets = append(buckets, l.NewLeakyBucket(capacity, rate, locker, backend, clock, s.logger))
+func (s *LimitersTestSuite) leakyBuckets(capacity int64, rate time.Duration, clock l.Clock) map[string]*l.LeakyBucket {
+	buckets := make(map[string]*l.LeakyBucket)
+	for lockerName, locker := range s.lockers(true) {
+		for backendName, backend := range s.leakyBucketBackends() {
+			buckets[lockerName+":"+backendName] = l.NewLeakyBucket(capacity, rate, locker, backend, clock, s.logger)
 		}
 	}
 	return buckets
 }
 
-func (s *LimitersTestSuite) leakyBucketBackends() []l.LeakyBucketStateBackend {
-	return []l.LeakyBucketStateBackend{
-		l.NewLeakyBucketInMemory(),
-		l.NewLeakyBucketEtcd(s.etcdClient, uuid.New().String(), time.Second, false),
-		l.NewLeakyBucketEtcd(s.etcdClient, uuid.New().String(), time.Second, true),
-		l.NewLeakyBucketRedis(s.redisClient, uuid.New().String(), time.Second, false),
-		l.NewLeakyBucketRedis(s.redisClient, uuid.New().String(), time.Second, true),
-		l.NewLeakyBucketMemcached(s.memcacheClient, uuid.New().String(), time.Second, false),
-		l.NewLeakyBucketMemcached(s.memcacheClient, uuid.New().String(), time.Second, true),
-		l.NewLeakyBucketDynamoDB(s.dynamodbClient, uuid.New().String(), s.dynamoDBTableProps, time.Second, false),
-		l.NewLeakyBucketDynamoDB(s.dynamodbClient, uuid.New().String(), s.dynamoDBTableProps, time.Second, true),
+func (s *LimitersTestSuite) leakyBucketBackends() map[string]l.LeakyBucketStateBackend {
+	return map[string]l.LeakyBucketStateBackend{
+		"LeakyBucketInMemory":   l.NewLeakyBucketInMemory(),
+		"LeakyBucketEtcd-":      l.NewLeakyBucketEtcd(s.etcdClient, uuid.New().String(), time.Second, false),
```

Comment:
Here and in other similar places: how about replacing "+" and "-" with smth like "WithRaceCheck" and "NoRaceCheck"?

Reply:
Sure, let me update the PR.